neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+                return false;
+            string operation = engine.CurrentContext.EvaluationStack.Pop().GetString();
+            VMArray args = (VMArray)engine.CurrentContext.EvaluationStack.Pop();
+            StackItem result;
+            switch (operation)
+            {
+                case "supportedStandards":
+                    result = SupportedStandards.Select(p => (StackItem)p).ToList();
+                    break;
+                case "name":
+                    result = Name;
+                    break;
+                case "symbol":
+                    result = Symbol;
+                    break;
+                case "decimals":
+                    result = Decimals;
+                    break;
+                case "totalSupply":
+                    result = TotalAmount;
+                    break;
+                case "balanceOf":
+                    result = BalanceOf(engine, args[0].GetByteArray());
+                    break;
+                case "transfer":
+                    result = Transfer(engine, args[0].GetByteArray(), args[1].GetByteArray(), args[2].GetBigInteger());
+                    break;
+                case "initialize":
+                    result = Initialize(engine);
+                    break;
+                case "unclaimedGas":
+                    result = UnclaimedGas(engine, args[0].GetByteArray(), (uint)args[1].GetBigInteger());
+                    break;
+                case "registerValidator":
+                    result = RegisterValidator(engine, args[0].GetByteArray());
+                    break;
+                case "vote":
+                    result = Vote(engine, args[0].GetByteArray(), ((VMArray)args[1]).Select(p => p.GetByteArray().AsSerializable<ECPoint>()).ToArray());
+                    break;
+                case "getValidators":
+                    result = GetValidators(engine).Select(p => (StackItem)p.ToArray()).ToArray();
+                    break;
+                default:
+                    return false;
+            }
+            engine.CurrentContext.EvaluationStack.Push(result);
+            return true;
+        }
+
+        private static StorageKey CreateStorageKey(UInt160 script_hash, byte prefix, byte[] key = null)
+        {
+            StorageKey storageKey = new StorageKey
+            {
+                ScriptHash = script_hash,
+                Key = new byte[sizeof(byte) + (key?.Length ?? 0)]
+            };
+            storageKey.Key[0] = prefix;
+            if (key != null)
+                Buffer.BlockCopy(key, 0, storageKey.Key, 1, key.Length);
+            return storageKey;
+        }
+
+        private static BigInteger BalanceOf(ApplicationEngine engine, byte[] account)
+        {
+            if (account.Length != 20) throw new ArgumentException();
+            StorageItem storage = engine.Service.Snapshot.Storages.TryGet(CreateStorageKey(ScriptHash, Prefix_Account, account));
```

Comment:
A StorageGet method should be provided on NativeContract base class, including the possibility to add a prefix... 
This avoid directly mentioning Snapshot here.

Reply:
We can't provide `StorageGet`, because we need `Add`, `TryGet`, `GetAndChange`.